src/logger/BaseLogger_test.go

```diff
@@ -0,0 +1,225 @@
+package logger
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"strings"
+)
+
+func TestCreatesInstance(t *testing.T) {
+	logger := NewBaseLoggerAgent(MOCK_AGENT)
+	assert.NotNil(t, logger)
+	assert.Equal(t, logger.getAgent(), MOCK_AGENT)
+	assert.false(t, logger.getEnableable())
+	assert.false(t, logger.getEnabled())
+	assert.Nil(t, logger.getQueue())
+	assert.Nil(t, logger.getUrl)
+}
+
+func TestCreatesMultipleInstances(t *testing.T) {
+	agent1 := "agent1"
+	agent2 := "agent2"
+	agent3 := "agent3"
+	url1 := "https://resuface.io"
+	url2 := "https://whatever.com"
+	helper := NewTestHelper()
+	logger1 := NewBaseLoggerAgentUrl(agent1, url1)
+	logger2 := NewBaseLoggerAgentUrl(agent2, url2)
+	logger3 := NewBaseLoggerAgentUrl(agent3, helper.demoURL)
+
+	assert.Equal(t, logger1.getAgent(), agent1)
+	assert.True(t, logger1.getEnableable())
+	assert.True(t, logger1.getEnabled())
+	assert.Equal(t, logger1.getUrl(), url1)
+
+	assert.Equal(t, logger2.getAgent(), agent2)
+	assert.True(t, logger2.getEnableable())
+	assert.True(t, logger2.getEnabled())
+	assert.Equal(t, logger2.getUrl(), url2)
+
+	assert.Equal(t, logger3.getAgent(), agent3)
+	assert.True(t, logger3.getEnableable())
+	assert.True(t, logger3.getEnabled())
+	assert.Equal(t, logger3.getUrl(), helper.demoURL)
+
+	//TODO: implement usage loggers
```

Comment:
usageloggers will most likely be a singelton so we will need a method to get it's pointer (i.e. `GetUsageLoggers()` or `UsageLoggers()`